angular/src/lib/input/input.directive.ts

```diff
@@ -0,0 +1,335 @@
+/** @component input */
+
+import {
+  DoCheck,
+  ElementRef,
+  Directive,
+  Input,
+  OnChanges,
+  OnInit,
+  OnDestroy,
+  SimpleChanges,
+  Optional,
+  Self,
+  HostBinding,
+} from '@angular/core';
+import { AutofillMonitor } from '@angular/cdk/text-field';
+import { takeUntil } from 'rxjs/operators';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { getSupportedInputTypes, Platform } from '@angular/cdk/platform';
+import {
+  InputGroupService
+} from '../input-group';
+import {
+  InputService
+} from './input.service';
+import { Subject } from 'rxjs';
+import {
+  ControlValueAccessor,
+  NgForm,
+  FormControl,
+  FormGroupDirective,
+  NgControl,
+} from '@angular/forms';
+
+let nextUniqueId = 0;
+
+// Invalid input type. Using one of these will throw an MatInputUnsupportedTypeError.
```

Comment:
um .. I don't think this comment is correct .. maybe it should be something besides the Material one?

Reply:
Will change to MDInputUnsupportedTypeError